### PR TITLE
Pilet setup on build 

### PR DESCRIPTION
**Pull request by DanteDeRuwe**
This change will make sure the `Piral.Blazor.Tools` package transforms a regular Blazor app into a pilet.

#### Workflow:
- user creates blazor app `My.Blazor.Pilet`
- user adds `<PiralInstance>app-shell-name</PiralInstance>` as a property in his `My.Blazor.Pilet.csproj`
- user adds `Piral.Blazor.Tools` nuget package

#### Upon build of the Blazor project, `Piral.Blazor.Tools` will: 
- scaffold a pilet using the piral cli (`npx pilet new app-shell-name`)
- copy over a `blazor.codegen` (and pass on targetframework and configuration)
- modify the `index.tsx` to make use of this codegen
  - ```ts
    import { PiletApi } from 'app-shell-name';
    import { registerDependencies, registerBlazorPages, registerBlazorExtensions, setupPilet } from './blazor.codegen';
    
    export function setup(app: PiletApi) {
        registerDependencies(app);
        registerBlazorPages(app);
        registerBlazorExtensions(app);
        setupPilet(app);
    }
      ``` 
- modify the Blazor project's `wwwroot/index.html` for use with a development proxy
- copy over a kras script injector (and pass on targetframework and configuration) to act as a development proxy for IISExpress! 
  - This allows breakpoints to be triggered if a debugger is attached 🎉 
- install the `Piral.Blazor.Analyzer` tool locally for use in `blazor.codegen`

Piral-specific files are added outside the solution directory, in a `piral~` directory (that is by default not checked in thanks to the default .NET `.gitignore` file)

The user can create a `setup.tsx`  in the solution dir to register components on the pilet API, that are later integrated in the `index.tsx`.

#### Still to be done or considered:
- running `pilet debug` when Blazor application is debugged.
- changing the target framework, configuration or app shell name once the pilet is already scaffolded won't change the codegen. For this the `piral~` folder should be deleted and a rebuild has to be done

### CODE REVIEW

File: src/Piral.Blazor.Tools/content/src/blazor.codegen [hunk 1]
```diff
@@ -200,6 +206,9 @@ module.exports = async function () {
     return `    
     ${definePathCode}
     module.exports = {
+      setupPilet : (app) => {
+        ${setupFileExists ? `import("${setupFilePath}").then(e=> e.default(app))` : ''};
```
Comment (FlorianRappl) [inline src/Piral.Blazor.Tools/content/src/blazor.codegen]: Why lazy load it? This is immediately required. Use `require('${setupFilePath}').default`.

```js
setupPilet: ${setupFileExists ? `require('${setupFilePath}').default` : '() => {}'
```

File: src/Piral.Blazor.Tools/content/src/blazor.codegen [hunk 2]
```diff
@@ -217,6 +226,9 @@ module.exports = async function () {
     return `
     ${definePathCode}
     module.exports = {
+      setupPilet : (app) => {
```
Comment (FlorianRappl) [inline src/Piral.Blazor.Tools/content/src/blazor.codegen]: Hm why this duplicate?

Make it reusable with a function.

File: src/Piral.Blazor.Tools/content/src/mocks/debug.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+const bbjsonPath = resolve(blazorprojectfolder, 'bin', configuration, targetFramework, 'wwwroot', '_framework', bbjson);
+
+const appshellname = require(resolve(piralPiletFolder, pjson)).piral.name;
+const appshellAppDir = resolve(require.resolve(appshellname), '..');
+
+const originalManifest = require(resolve(appshellAppDir, '_framework', bbjson));
+const piletManifest = require(bbjsonPath);
+
+const launchSettingsPath = resolve(blazorprojectfolder, 'Properties', lsjson);
+const iisUrl = require(launchSettingsPath)?.iisSettings?.iisExpress?.applicationUrl;
```
Comment (FlorianRappl) [inline src/Piral.Blazor.Tools/content/src/mocks/debug.js]: This wouldn't work in older Node versions (e.g., Node 10 and Node 12). Please write JS that works with Node 10 (or at least 12).

**Review by FlorianRappl — COMMENTED** [review]
Looks great!

File: src/Piral.Blazor.Tools/content/src/mocks/debug.js (hunk 3 above)
Comment (DanteDeRuwe) [reply src/Piral.Blazor.Tools/content/src/mocks/debug.js]: That was my C# (and possibly TS) brain talking. Good catch. Changed it!

**Review by FlorianRappl — APPROVED** [review]

**Pull request merged**